src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -30,21 +30,33 @@ internal class UserDefinedFunction : TexlFunction
     {
         private readonly bool _isImperative;
         private readonly IEnumerable<UDFArg> _args;
-        private TexlBinding _binding;
+        private readonly UserDefinedFunctionLibrary _library;
```

Comment:
>UserDefinedFunctionLibrary [](http://example.com/codeflow?start=25&length=26)

this adds a level of indirection, but can we remove the dependency entirely?
Ie, if we're just doing intellisense at the callsite, we don't need this. 
And if we're doing codegen , we should already have it through a separate channel. 

(this begs the issue for IsAsync)

Reply:
Makes sense, the only reason we have reference binding/library is to derive the value for IsAsync as it is used in several places in the PAClient.